fetch-array-buffer/index.html

```diff
@@ -23,59 +20,55 @@ <h1>Fetch arrayBuffer example</h1>
     <pre></pre>
   </body>
   <script>
-    // define variables
-    var audioCtx = new (window.AudioContext || window.webkitAudioContext)();
-    var source;
-
-    var pre = document.querySelector('pre');
-    var myScript = document.querySelector('script');
-    var play = document.querySelector('.play');
-    var stop = document.querySelector('.stop');
-    var errorDisplay = document.querySelector('.error');
+    const pre = document.querySelector("pre");
+    const myScript = document.querySelector("script");
+    const play = document.querySelector(".play");
+    const stop = document.querySelector(".stop");
+    const errorDisplay = document.querySelector(".error");
 
     // use fetch to load an audio track, and
     // decodeAudioData to decode it and stick it in a buffer.
     // Then we put the buffer into the source
     function getData() {
```

Comment:
What about:

```js
function getData() {
  const audioCtx = new AudioContext();
  return fetch("viper.ogg")
    .then((response) => {
      if (!response.ok) {
        throw new Error(`HTTP error, status = ${response.status}`);
      }
      return response.arrayBuffer();
    })
    .then((buffer) => audioCtx.decodeAudioData(buffer))
    .then((decodedData) => {
      const source = audioCtx.createBufferSource();
      source.buffer = decodedData;
      source.connect(audioCtx.destination);
      return source;
    });
}
```

Not familiar with web APIs so purely JS-based refactor from the original code.